lms/djangoapps/support/views/enrollments.py

```diff
@@ -84,6 +84,55 @@ def get(self, request, username_or_email):
 
     @method_decorator(require_support_permission)
     def post(self, request, username_or_email):
+        """Allows support staff to create a user's enrollment."""
+        try:
+            course_id = request.data['course_id']
+            course_key = CourseKey.from_string(course_id)
+            mode = request.data['mode']
+            enrollment_modes = [
+                enrollment_mode['slug']
+                for enrollment_mode in self.get_course_modes(course_key)
+            ]
+            if mode not in enrollment_modes:
+                return HttpResponseBadRequest(
+                    f'{str(mode)} is not a valid mode for {str(course_id)}. '
+                    f'Possible valid modes are {str(enrollment_modes)}'
+                )
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_key)
+            if enrollment is not None:
+                return HttpResponseBadRequest(
+                    f'The user {str(username_or_email)} is already enrolled in {str(course_id)}.'
+                )
+            reason = request.data['reason']
+            enrollment = CourseEnrollment.enroll(
+                user=user, course_key=course_key, mode=mode
+            )
+        except KeyError as err:
+            return HttpResponseBadRequest(f'The field {str(err)} is required.')
+        except InvalidKeyError:
+            return HttpResponseBadRequest('Could not parse course key.')
+        except User.DoesNotExist:
+            return HttpResponseBadRequest(
+                'Could not find user {username}.'.format(
+                    username=username_or_email
+                )
+            )
+
+        # Wrapped in a transaction so that we can be sure the
+        # ManualEnrollmentAudit record is always created correctly.
+        with transaction.atomic():
+            manual_enrollment = ManualEnrollmentAudit.create_manual_enrollment_audit(
+                request.user,
+                enrollment.user.email,
+                ENROLLED_TO_ENROLLED,
```

Comment:
How come this transaction is from enrolled to enrolled? This view is creating an enrollment if it is not already present.

Reply:
Oh, It was meant to be `UNENROLLED_TO_ENROLLED`. My bad.